merge_sort/merge_sort.py

```diff
@@ -0,0 +1,68 @@
+def merge(array, left, right):
+    """
+    Perform Merge Operation between arrays.
+    Time Complexity: Theta(nLogn)
+    Auxiliary Space: O(n)
+    :param array: Iterable of elements
+    :param left: left limit for merge sort
+    :param right: right limit for merge sort
+    :return: no returns, merges arrays.
+    """
+    mid = int((left + right) / 2)
+    s1 = int(mid - left + 1)
+    s2 = int(right - mid)
+    l = [0] * s1
+    r = [0] * s2
+    for i in range(s1):
+        l[i] = array[int(left + i)]
+
+    for j in range(s2):
+        r[j] = array[mid + 1 + j]
+
```

Comment:
You will never need to access index if you work with truth value of `l` and just use `pop()` operations.